script/components-json/build.ts

```diff
@@ -45,5 +99,59 @@ fs.writeFileSync('generated/components.json', JSON.stringify(data, null, 2))
 // Print success message
 // eslint-disable-next-line no-console
 console.log(
-  `Successfully built "generated/components.json" from ${jsonFiles.length} files matching "src/**/*.docs.json"`,
+  `Successfully built "generated/components.json" from ${docsFiles.length} files matching "src/**/*.docs.json"`,
 )
+
+// -----------------------------------------------------------------------------
+// Helper functions
+// -----------------------------------------------------------------------------
+
+/**
+ * Returns an object mapping story names to their source code
+ *
+ * @example
+ * getStorySourceCode('src/components/Button/Button.stories.tsx')
+ * // {Default: '<Button>Button</Button>'}
+ */
+function getStorySourceCode(filepath: string) {
+  if (!fs.existsSync(filepath)) {
+    return {}
+  }
+
+  const code = fs.readFileSync(filepath, 'utf-8')
+
+  const ast = parse(code, {
+    sourceType: 'module',
+    plugins: ['jsx', 'typescript'],
+  })
+
+  const stories: Record<string, string> = {}
+
+  traverse(ast, {
+    ExportNamedDeclaration(path) {
+      const varDecloration = path.node.declaration as VariableDeclaration
+      const id = varDecloration.declarations[0].id as Identifier
+      const name = id.name
+      const func = varDecloration.declarations[0].init as ArrowFunctionExpression
+
+      const code = prettier
+        .format(generate(func).code, {
+          parser: 'typescript',
+          singleQuote: true,
+          trailingComma: 'all',
+          semi: false,
+        })
+        .trim()
+        .replace(/;$/, '')
```

Comment:
This might cover the `;` usage at the beginning too that seems to be showing up 🤔 

```suggestion
        .replace(/;$/, '')
        .replace(/^;/, '')
```

Reply:
Good catch. Done ✅